crates/common/certificate/src/parse_root_certificate.rs

```diff
@@ -27,27 +29,65 @@ pub fn create_tls_config(
         .with_single_cert(cert_chain, pvt_key)?)
 }
 
-fn new_root_store(cert_path: PathBuf) -> Result<RootCertStore, CertificateError> {
+pub fn add_certs_from_file(
+    root_store: &mut RootCertStore,
+    cert_file: impl AsRef<Path>,
+) -> Result<(), CertificateError> {
+    let cert_chain = read_cert_chain(cert_file)?;
+    for cert in cert_chain {
+        root_store
+            .add(&cert)
+            .map_err(|_| CertificateError::RootStoreAdd)?;
+    }
+
+    Ok(())
+}
+
+pub fn add_certs_from_directory(
+    root_store: &mut RootCertStore,
+    cert_dir: impl AsRef<Path>,
+) -> Result<(), CertificateError> {
+    let files = fs::read_dir(cert_dir)?;
+    let certs = files.filter_map(|f| f.ok()).filter(|file| {
+        file.path()
+            .extension()
+            .filter(|&extension| {
+                ["pem", "cer", "crt"]
```

Comment:
This is okay. However, such a closed list of accepted names is always a source of surprises. For instance, I would expect to see `cert` but not `cer`.

Reply:
These 3 extensions are what MQTT CLI uses (`mosquitto_pub/sub` only allows `.crt`), but it's largely a matter of convention. More complete list of extensions can be found [here](https://en.wikipedia.org/wiki/X.509#Certificate_filename_extensions), and maybe we could accept more extensions, but I don't think it's an issue as long as we properly document the extensions we accept, which I've just noticed I forgot about. Will add the list of accepted extensions to the description of relevant configuration options.

Reply:
Okay. So let's start with these 3 extensions.